source/intercept/source/collect/Intercept.cc

```diff
@@ -38,22 +39,56 @@ namespace fs = std::filesystem;
 
 namespace {
 
-    rust::Result<ic::Execution> capture_execution(const flags::Arguments& args, sys::env::Vars &&environment)
+    rust::Result<ic::Configuration>
+    into_configuration(const flags::Arguments &args) {
+        auto output_file_arg = args.as_string(cmd::intercept::FLAG_OUTPUT);
+        auto library_arg = args.as_string(cmd::intercept::FLAG_LIBRARY);
+        auto wrapper_arg = args.as_string(cmd::intercept::FLAG_WRAPPER);
+        auto wrapper_dir_arg = args.as_string(cmd::intercept::FLAG_WRAPPER_DIR);
+        auto verbose_arg = args.as_bool(flags::VERBOSE);
+        auto force_preload_arg = args.as_bool(cmd::intercept::FLAG_FORCE_PRELOAD);
+        auto force_wrapper_arg = args.as_bool(cmd::intercept::FLAG_FORCE_WRAPPER);
+        auto command_arg = args.as_string_list(cmd::intercept::FLAG_COMMAND);
+
+        ic::Configuration config;
```

Comment:
For now the configuration is default constructed (it's initialized with default values), but it would be better that the user can supply his own configuration (this will be done in a subsequent PR).